libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -996,3 +1055,61 @@ export interface SimulatorServerResponse {
   /** The error that occurred during the method call. */
   readonly error?: any;
 }
+
+function planUpdate(current: BaseResourceSchema[], next: BaseResourceSchema[]) {
```

Comment:
Does this need to handle dependencies? Like, if A depends on B and A needs to be updated, doesn't that mean B may need to be updated too?

Reply:
Yes, absolutely and definitely. I felt I missed something and originally thought about this as an issue but missed it.

The challenge here is that our simulator doesn't have a static dependency model.

Let me see what I can come up with.

Reply:
OK, added an execution graph. Should cover this.